src/GitHub.InlineReviews/InlineReviewsPackage.cs

```diff
@@ -1,24 +1,32 @@
 ﻿using System;
+using System.ComponentModel.Design;
 using System.Runtime.InteropServices;
+using System.Threading;
 using GitHub.InlineReviews.Commands;
 using GitHub.InlineReviews.Views;
 using GitHub.VisualStudio;
+using Microsoft.VisualStudio.ComponentModelHost;
 using Microsoft.VisualStudio.Shell;
 using Microsoft.VisualStudio.Shell.Interop;
+using Task = System.Threading.Tasks.Task;
 
 namespace GitHub.InlineReviews
 {
-    [PackageRegistration(UseManagedResourcesOnly = true)]
+    [PackageRegistration(UseManagedResourcesOnly = true, AllowsBackgroundLoading = true)]
     [Guid(Guids.InlineReviewsPackageId)]
     [ProvideAutoLoad(UIContextGuids80.SolutionExists)]
     [ProvideMenuResource("Menus.ctmenu", 1)]
     [ProvideToolWindow(typeof(PullRequestCommentsPane), DocumentLikeTool=true)]
-    public class InlineReviewsPackage : Package
+    public class InlineReviewsPackage : AsyncPackage
     {
-        protected override void Initialize()
+        protected override async Task InitializeAsync(
+            CancellationToken cancellationToken,
+            IProgress<ServiceProgressData> progress)
         {
-            base.Initialize();
-            PackageResources.Register(this);
+            await ThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
+            var menuService = (IMenuCommandService)(await GetServiceAsync(typeof(IMenuCommandService)));
+            var componentModel = (IComponentModel)(await GetServiceAsync(typeof(SComponentModel)));
+            PackageResources.Register(this, componentModel.DefaultExportProvider, menuService);
```

Comment:
It would be better to go through `GitHubServiceProvider` for this. `GitHubServiceProvider` already loads up `SComponentModel` when it gets initialized and uses it for lookups, guaranteeing that only types that are ours get returned in MEF queries.

Going through `GitHubServiceProvider` also means that we can bypass the `ExportProvider` look up at any time by precaching instances in it ahead of time (which is handy for tests and also if we want to avoid a MEF query for some reason).

Reply:
The problem with `GitHubServiceProvider` is that it doesn't support async lookups, and given that we're in `InitializeAsync` I figured that we really should be doing these async. Or have I missed something?